### PR TITLE
add Image controller

### PR DESCRIPTION
**Pull request by outscale-hmi**
Add Image Controller

### CODE REVIEW

File: controllers/oscmachine_image_controller_unit_test.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+			},
+			expImageFound:        false,
+			expReconcileImageErr: fmt.Errorf("GetImage generic error"),
+		},
+	}
+	for _, itc := range imageTestCases {
+		t.Run(itc.name, func(t *testing.T) {
+			_, machineScope, ctx, mockOscimageInterface := SetupWithImageMock(t, itc.name, itc.spec, itc.machineSpec)
+			imageSpec := itc.machineSpec.Node.Image
+			imageName := itc.machineSpec.Node.Image.Name + "-uid"
```
Comment (outscale-vbr) [inline controllers/oscmachine_image_controller_unit_test.go:199]: No uid in Name ?

File: controllers/oscmachine_image_controller_unit_test.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+			imageId := "omi-" + imageName
+			imageRef := machineScope.GetImageRef()
+			imageRef.ResourceMap = make(map[string]string)
+			imageRef.ResourceMap[imageName] = imageId
+			image := osc.ReadImagesResponse{
+				Images: &[]osc.Image{
+					{
+						ImageName: &imageName,
```
Comment (outscale-vbr) [inline controllers/oscmachine_image_controller_unit_test.go:207]: ImageId instead ?

Comment (outscale-vbr) [reply controllers/oscmachine_image_controller_unit_test.go:207]: To search ?

File: cloud/services/compute/image.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	oscAuthClient := s.scope.GetAuth()
+	readImageResponse, httpRes, err := oscApiClient.ImageApi.ReadImages(oscAuthClient).ReadImagesRequest(readImageRequest).Execute()
+	if err != nil {
+		fmt.Printf("Error with http result %s", httpRes.Status)
+		return nil, err
+	}
+	if len(readImageResponse.GetImages()) == 0 {
+		return nil, nil
+	}
+	image := readImageResponse.GetImages()[0]
```
[the remark this answers is not in the document]
Comment (outscale-hmi) [reply cloud/services/compute/image.go:29]: It's the same thing GetImage return the value or nil if not set and GetImageOK will return (*[]Image, bool),

File: controllers/oscmachine_image_controller.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+	infrastructurev1beta1 "github.com/outscale-dev/cluster-api-provider-outscale.git/api/v1beta1"
+	"github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/scope"
+	"github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/services/compute"
+	osc "github.com/outscale/osc-sdk-go/v2"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// ValidateTagNameValue check that tag name value is a valide name
+func ValidateImageName(imageName string) (string, error) {
```
[the remark this answers is not in the document]
Comment (outscale-hmi) [reply controllers/oscmachine_image_controller.go:17]: yes

**Review by outscale-vbr — APPROVED** [review]
Ok when test

**Pull request closed without merging**